backend/internal/model/game.go

```diff
@@ -208,6 +208,15 @@ func (g *game) IncrementHeatmap(xCord float64, yCord float64) error {
 	if y > heatmapUpperBound || y < 0 {
 		return errors.New("y ball position index out of range")
 	}
+	for i := x - 1; i <= x+1; i++ {
+		for j := y - 1; j <= y+1; j++ {
+			if i > 0 && i < heatmapUpperBound {
+				if j > 0 && j < heatmapUpperBound {
+					g.gameData.Heatmap[i][j]++
+				}
+			}
+		}
+	}
```

Comment:
Maybe we could check before iterating if we are on the edge of the heatmap to avoid checking every time this scenario in the loop. In most of the cases we won't be on the edge.
```suggestion
	if y == heatmapUpperBound || x == heatmapUpperBound || y == 0 || x == 0 {
		for i := x - 1; i <= x+1; i++ {
			for j := y - 1; j <= y+1; j++ {
				if i > 0 && i < heatmapUpperBound {
					if j > 0 && j < heatmapUpperBound {
						g.gameData.Heatmap[i][j]++
					}
				}
			}
		}
	} else {
		for i := x - 1; i <= x+1; i++ {
			for j := y - 1; j <= y+1; j++ {	
				g.gameData.Heatmap[i][j]++
			}
		}
	}
```

Reply:
That should work better, I'll commit this suggestion